fix: notebook fixes for notebook 1 and 3

daniel-rdt opened the pull request:
Closes #27.

This PR proposes the following fixes:
- [x] PyPSA v1.0.5 compatibility for workshop notebook 1 and 2
- [x] Fix error in notebook 1 solution by saving n1 ealier and then reloading it again for the solution
- [x] Version open-tyndp data packages, to differentiate between workshop 2 and 3 packages
- [x] Fix the `helpers` to a specific commit for workshop notebook 3
- [x] Update `environment.yaml`: remove pypsa lock

daniel-rdt committed:
feat: add PyPSA v1 compatibility and fix solution error by savind and reloading n1 network

daniel-rdt committed:
feat: add PyPSA v1 compatibility for notebook 2

daniel-rdt committed:
feat: version retrieved pre-built open-tyndp repos

daniel-rdt committed:
feat: fix helpers script retrieve to specific commit for workshop 3

daniel-rdt committed:
feat: remove open-tyndp directories from data at the end of the notebook to clean up

daniel-rdt commented:
Hey @tgilon. I added all the fixes and it is ready for review for you. It is now launching the GitHub page again without any errors

daniel-rdt commented:
Note: Testing of the updated notebooks on Colab was successful.

tgilon committed:
chore: remove locked pypsa version

tgilon reviewed (changes requested):
Thank you @daniel-rdt for these fixes. They will help users to have a more integrated experience when testing things across the different notebooks.

I tested all three notebooks locally using both the `environment.yaml` and `pip`. Both worked as expected.

Here below my comments:

- [x] Could you please fix the version of all the retrieved files that currently refer to master? It's working with the current version, so fixing the current commit should be fine.
- [x] Could you also fix the image in the first notebook? It refers to an image that no longer exists.
    ```
    <img src="https://github.com/fneum/data-science-for-esm/raw/main/data-science-for-esm/pypsa-logo.png" width="300px" />
    ```
- [x] Could you add a note to the description stating that all the notebooks work with PyPSA v1.0.5?

- [x] Here is a suggestion for detecting whether the CI is running the workflow?

    ```python
        import shutil
        import os
    
        # Only clean up data when running in CI environment
        if os.getenv("CI"):
            rm_dir = "data/open-tyndp-20251129"
            print(f"CI environment detected. Cleaning up notebook data by removing {rm_dir}.")
            shutil.rmtree(rm_dir, ignore_errors=True)
        else:
            print("Skipping cleanup (not in CI environment).")
    ```
    
    Without this, the current rule that removes the ZIP at the end breaks the logic of the downloader. If you run all twice, you will actually download the data twice.

daniel-rdt committed:
feat: update urls and fix retrieves of files to current commit

daniel-rdt committed:
fix: fix PyPSA logo in notebook 1

daniel-rdt committed:
doc: add note about PyPSA v1.0.5 compatibility

daniel-rdt committed:
ci: only clean up data when running in CI env

daniel-rdt commented:
@tgilon I've addressed your comments and the newest notebook version is launched in preview. Maybe you can have a final look and we can merge if you're happy

tgilon commented on open-tyndp-workshops/20250409-workshop-pypsa-01.ipynb:
```diff
@@ -24,7 +24,7 @@
    "metadata": {},
    "source": [
-    "<img src=\"https://github.com/fneum/data-science-for-esm/raw/main/data-science-for-esm/pypsa-logo.png\" width=\"300px\" />\n",
+    "<img src=\"pypsa-logo.png\" width=\"300px\" />\n",
```
The file in missing.

daniel-rdt replied:
Where and when is it missing? It is there in the [repo](https://github.com/open-energy-transition/open-tyndp-workshops/blob/792b8474ab5096e5ab8db2822af4fcd9fe659eb6/open-tyndp-workshops/pypsa-logo.png). Or is it a problem on colab?

tgilon commented on open-tyndp-workshops/20250624-workshop-pypsa-02.ipynb:
```diff
@@ -2309,16 +2308,23 @@
+    "\n",
+    "# Only clean up data when running in CI environment\n",
+    "if os.getenv(\"CI\"):\n",
+    "    rm_dir = \"open-tyndp-20251016\"\n",
```
I prefer the version of nb 3 with `data/`

tgilon reviewed (commented):
Tks, just two tiny suggestions before merging!

daniel-rdt committed:
change clean up directory for notebook 2

daniel-rdt committed:
change paths for pypsa and snakemake logos

daniel-rdt committed:
change paths for pypsa and snakemake logos ii

tgilon reviewed (approved):
GTG, tks!
